src/main/java/io/xpring/codec/addresses/XAddressCodec.java

```diff
@@ -0,0 +1,126 @@
+package io.xpring.codec.addresses;
+
+import static com.google.common.io.BaseEncoding.base16;
+
+import com.google.common.base.Preconditions;
+import io.xpring.codec.base58.B58;
+import io.xpring.codec.basex.IBaseX;
+import io.xpring.xrpl.ClassicAddress;
+
+import java.util.Optional;
+
+/**
+ * Encodes and decodes X-Address as defined by the spec: https://github.com/xrp-community/standards-drafts/issues/6.
+ */
+public class XAddressCodec {
+
+  private static final int XADDRESS_EXPECTED_LENGTH = 30;
+  private static final int ACCOUNT_ID_EXPECTED_LENGTH = 20;
+  private static final int ACCOUNT_ID_VERSION = 0;
+  private static final String TAG_PRESENT_FLAG = "01";
+  private static final long MAX_TAG = (long) Math.pow(2, 32) - 1;
+  // XRPL uses Base58 but with an alphabet in a peculiar order (see https://xrpl.org/base58-encodings.html)
+  private static B58 codec = new B58("rpshnaf39wBUDNEGHJKLM4PQRST7VWXYZ2bcdeCg65jkm8oFqi1tuvAxyz");
+  private static String RESERVED_SUFFIX = "00000000";
+
+  /**
+   * Encodes a classic address to an x-address.
+   *
+   * @param classicAddress the address to encode.
+   * @return encoded x-address.
+   */
+  public static String encode(ClassicAddress classicAddress) {
+    // To compute the X-Address, several hex values are computed and concatenated together, then encoded
+    // using Base58 with a XRPL version
+    try {
+      Prefix prefix = classicAddress.isTest() ? Prefix.TESTNET : Prefix.MAINNET;
+      String addressBytes = base16().encode(ClassicAddressCodec.decodeAccountID(classicAddress.address()));
+      String tagFlag = classicAddress.tag().isPresent() ? TAG_PRESENT_FLAG : "00";
+      String tagBytes = encodeTag(classicAddress.tag());
+      byte [] xaddress = base16().decode(prefix.prefix + addressBytes + tagFlag + tagBytes + RESERVED_SUFFIX);
```

Comment:
optional nit: based on capitalization of  the class I would refer to this as `xAddress`

Reply:
I had `xAddress` originally but then it failed the checkstyle rules so I changed it.

Reply:
OK, I see that you had been suppressing that warning in other classes that use a variable name `xAddress`. I'll follow suit here as well with the suppress warning.